Legacy Date Expressions

jsherer opened the pull request:
Added support for legacy date expressions like:

`+5d`
`+12h`
`-10m`

jsherer commented:
#9

jsherer commented:
What do you think about this one too @bcooksey?

bcooksey commented:
@jsherer if we want to be complete, here is the matcher we should use:

```python
        d = re.match(r'^((?P<years>(\d+))y(ear(s)?)?)?'
                     r'((?P<months>(\d+))mo(nth(s)?)?)?'
                     r'((?P<weeks>(\d+))w((ee)?(k(s)?)?))?'
                     r'((?P<days>(\d+))d(ay(s)?)?)?'
                     r'((?P<hours>(\d+))h(ou)?(r(s)?)?)?'
                     r'((?P<minutes>(\d+))m(in(ute)?(s)?)?)?'
                     r'((?P<seconds>(\d+))s(ec(ond)?(s)?)?)?$',
                     smart_unicode(word))
```

jsherer commented:
Not too many missing from our current scheme. I'll update this to hit these:
```
1y
1year
1years
1mo
1month
1months
1w
1wk
1wks
1week
1weeks
1d
1day
1days
1h
1hr
1hrs
1hour
1hours
1m
1min
1mins
1minute
1minutes
1s
1sec
1secs
1second
1seconds
```

bcooksey commented:
Oh, cool if our existing code covers a lot of these, I didn't glean that from my read through. Maybe that's a +1 to the formatting of the regex I pasted; it helped me see the different permutations we support.

jsherer commented:
Added tests for these cases and updated the regex to be more readable. Changed it up a bit to fit our docs to provide negative time deltas alongside positive time deltas, ala `+5 days - 1 hour`.

@bcooksey

bcooksey commented:
Yea, looks good; and I love the test.
